drivers/usb/device/usb_dc_rpi_pico.c

```diff
@@ -117,11 +123,85 @@ static int udc_rpi_start_xfer(uint8_t ep, const void *data, size_t len)
 	return 0;
 }
 
+/*
+ * @brief get a bit mask associated with an endpoint address
+ *
+ * This function converts a zephyr endpoint address into a
+ * bit mask that can be used with registers:
+ *  - buf_status
+ *  - buf_cpu_should_handle
+ *  - abort
+ *  - abort_done
+ *  - ep_nak_stall_status
+ *
+ * @param [in] ep        Endpoint address
+ *
+ * @return a 32-bit mask specific to rpi_pico hardware
+ */
+static inline uint32_t udc_rpi_endpoint_mask(const uint8_t ep)
+{
+	const int bit_index = (USB_EP_GET_IDX(ep) << 1) | !!USB_EP_DIR_IS_OUT(ep);
+
+	return BIT(bit_index);
+}
+
+/*
+ * @brief Cancel an already armed endpoint
+ *
+ * @param [in] ep        Endpoint address
+ */
+static void udc_rpi_cancel_endpoint(const uint8_t ep)
+{
+	struct udc_rpi_ep_state *const ep_state = udc_rpi_get_ep_state(ep);
+
+	if (*ep_state->buf_ctl & USB_BUF_CTRL_AVAIL) {
+		const uint32_t mask = udc_rpi_endpoint_mask(ep);
+
+		hw_set_alias(usb_hw)->abort = mask;
+		while ((usb_hw->abort_done & mask) != mask)
+			;
+		*ep_state->buf_ctl &= ~USB_BUF_CTRL_AVAIL;
+		hw_clear_alias(usb_hw)->abort = mask;
+	}
+}
+
 static void udc_rpi_handle_setup(void)
 {
+	const struct udc_rpi_ep_state *const ep_state = udc_rpi_get_ep_state(USB_CONTROL_EP_OUT);
 	struct cb_msg msg;
 
-	state.setup_available = true;
+	/* Normally all control transfers should complete before a new setup transaction is
+	 * sent, however in some rare cases from the perspective of the device, a new setup
+	 * transaction could arrive prematurely, in which case the previous control transfer
+	 * should be aborted. See section 5.5.5 of the Universal Serial Bus Specification,
+	 * version 2.0.
```

Comment:
DONE, I think. I have shortened the comments, however could you clarify what the exact rule is? Should indentation be included in those 80 characters? If yes, then how many characters should tab be treated as? 1/2/4/8? Should comments be more restricted in width than code below and above them? I was assuming before that my comment width was okay, because code around it was wider, and the PR checker didn't complain about any of it. It would be helpful if all of these rules were checked by the PR checker, then we wouldn't have to waste time on communication about this simple issue.